homeassistant/components/knx/config_flow.py

```diff
@@ -104,35 +110,41 @@ async def async_step_type(self, user_input: dict | None = None) -> FlowResult:
         """Handle connection type configuration."""
         if user_input is not None:
             connection_type = user_input[CONF_KNX_CONNECTION_TYPE]
-            if connection_type == CONF_KNX_AUTOMATIC:
-                entry_data = DEFAULT_ENTRY_DATA | KNXConfigEntryData(
-                    connection_type=CONF_KNX_AUTOMATIC
-                )
-                return self.async_create_entry(
-                    title=CONF_KNX_AUTOMATIC.capitalize(),
-                    data=entry_data,
-                )
-
             if connection_type == CONF_KNX_ROUTING:
                 return await self.async_step_routing()
 
-            if connection_type == CONF_KNX_TUNNELING and self._found_tunnels:
+            if connection_type == CONF_KNX_TUNNELING:
+                self._found_tunnels = [
+                    gateway
+                    for gateway in self._found_gateways
+                    if gateway.supports_tunnelling
```

Comment:
Unfortunately the lambda doesn't mypy then because `GatewayDescriptor.individual_address` is still seen optional. (and sorting by string representation is awkward as eg. "15.15.255" comes before "2.0.0")